src/pages/workspaces/workspace/analysis/modals/persistent-disk-controls.ts

```diff
@@ -77,33 +102,39 @@ export const PersistentDiskSection = ({ diskExists, computeConfig, updateCompute
       label({ style: computeStyles.label }, ['Persistent disk']),
       div({ style: { marginTop: '0.5rem' } }, [
         'Persistent disks store analysis data. ',
-        h(Link, { onClick: handleLearnMoreAboutPersistentDisk }, ['Learn more about persistent disks and where your disk is mounted.'])
+        h(Link, {
+          onClick: () => handleLearnMoreAboutPersistentDisk({ setViewMode })
+        }, ['Learn more about persistent disks and where your disk is mounted.'])
       ]),
       div({ style: { ...gridStyle, gridGap: '1rem', gridTemplateColumns: '15rem 5.5rem', marginTop: '0.75rem' } }, [
-        diskExists ?
-          h(TooltipTrigger, {
-            content: [
-              'You already have a persistent disk in this workspace. ',
-              'Disk type can only be configured at creation time. ',
-              'Please delete the existing disk before selecting a new type.'
-            ],
-            side: 'bottom'
-          }, [h(PersistentDiskType, { diskExists, computeConfig, updateComputeConfig })]) : h(PersistentDiskType, { diskExists, computeConfig, updateComputeConfig }),
+        // TODO: we inconsistently use GCP and Gcp, once cloudPlatform is typed, make stronger comparison here
+        ['GCP', 'Gcp'].includes(cloudPlatform) ? diskType({ persistentDiskExists, computeConfig, updateComputeConfig }) : false,
         h(div, [
           label({ htmlFor: diskSizeId, style: computeStyles.label }, ['Disk Size (GB)']),
-          div({ style: { marginTop: '0.5rem' } }, [
+          div({ style: { width: 75, marginTop: '0.5rem' } }, [
             h(NumberInput, {
               id: diskSizeId,
               min: 10,
               max: 64000,
               isClearable: false,
               onlyInteger: true,
-              value: computeConfig.selectedPersistentDiskSize,
-              onChange: updateComputeConfig('selectedPersistentDiskSize')
+              value: computeConfig.persistentDiskSize,
+              onChange: updateComputeConfig('persistentDiskSize')
             })
           ])
         ])
       ])
     ])
   ])
 }
+
+const diskType = ({ persistentDiskExists, computeConfig, updateComputeConfig }) => {
```

Comment:
This signature does not follow the name casing convention for functional components, and overloads the name `diskType`, which is also a field in DiskConfig. Also, should probably have typed props. Frankly, I'm starting to think this PR conflates a `DiskConfig` with a `ComputeConfig`. Do components that deal with disks really need to know about the entire `RuntimeConfig`?

Reply:
For Azure, the disk configuration is currently just diskSize, everything else is a default on Leo's side.
It's also top-level.
While they don't necessarily need to be tied together anymore, it did make sense at the time.